server/types/models/video/video-channel-sync.ts

```diff
@@ -0,0 +1,12 @@
+import { VideoChannelSyncModel } from "@server/models/video/video-channel-sync"
+import { FunctionProperties, PickWith } from "@shared/typescript-utils"
+import { MChannelFormattable } from "./video-channels"
+
+type Use<K extends keyof VideoChannelSyncModel, M> = PickWith<VideoChannelSyncModel, K, M>
+
+export type MChannelSync = VideoChannelSyncModel
```

Comment:
`export type MChannelSync = Omit<VideoChannelSyncModel, 'VideoChannel'>`

Reply:
Not sure how to do that properly:
1. I would like to keep the video-name attribute for audit logging: https://github.com/fflorent/PeerTube/blob/2866df4eb4b5fa395fc0bc8e60c7007507cc3d34/server/helpers/audit-logger.ts#L263-L273
2. The `res.locals.videoChannelSync` (stored as `syncInstance` is of type `MChannelSync` and I use it to pass to the AuditLogger: thttps://github.com/fflorent/PeerTube/blob/2866df4eb4b5fa395fc0bc8e60c7007507cc3d34/server/helpers/audit-logger.ts#L263-L273
3. But I must pass an instance which has a `VideoChannel` property.

I wonder what could be a good way to bring this change and make the code compile.

Some options I see:
1. Casting `syncInstance` to a `VideoChannelSyncFormattable` (is it safe?)
2. Make `res.locals.videoChannelSync` of type `VideoChannelSyncFormattable`

Help welcome on this one. Thanks in advance

Reply:
update `res.locals.videoChannelSync` to be `MChannelSyncChannel` (so channel sync + channel association).

Reply:
Done in [515dae8](https://github.com/Chocobozzz/PeerTube/pull/5135/commits/515dae816fe50209aa19e39ac683d28ce3e6b8fd)